.github/workflows/release.yml

```
@@ -3,22 +3,17 @@ name: Release
 on:
   release:
     types: [published]
+  push:
+  pull_request:
 
 jobs:
   release:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-
-    - name: Set up Python
-      uses: actions/setup-python@v5
       with:
-        python-version: '3.9'
-
-    - name: Install Python dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install wheel twine
+        persist-credentials: false
+    - uses: astral-sh/setup-uv@v5
 
     - name: Verify tag is documented
```

Should run only for release build:

```suggestion
    - name: Verify tag is documented
      if: github.event_name == 'release'
```